database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -58,6 +71,10 @@ private void init(Query query) {
         mQuery = query;
     }
 
+    void setPreChangeEventListener(PreChangeEventListener listener) {
```

Comment:
Not sure what to do about this api... seems like other devs might want to use it, but it's kinda gross.